File: js/367. Valid Perfect Square.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function(num) {
    for (let i = 1; i <= num; i++) {
        let tmp = i * i;
        if (tmp === num) {
            return true;
        }
        else if (tmp > num) {
            return false;
        }
    }
    return false;
};

File: go/367. Valid Perfect Square.go
package main
func isPerfectSquare(num int) bool {
	for i := 1; i <= num; i++ {
		tmp := int64(i) * int64(i)
		if tmp == int64(num) {
			return true
		}
		if tmp > int64(num) {
			return false
		}
	}
	return false
}
